packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -507,13 +531,14 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
             const { summaryVersion } = await summarize(summarizer);
             const dataObject = await getTombstonedDataObjectFromSummary(summaryVersion, unreferencedId);
 
-            // Note: if a user makes a request that looks like this, we will also think that the request is via handle
-            const request: IRequest = { url: unreferencedId, headers: { [RuntimeHeaders.viaHandle]: true } };
-            const response = await dataObject._context.IFluidHandleContext.resolveHandle(request);
+            const serializer = new FluidSerializer(dataObject._context.IFluidHandleContext, () => {});
+            const handle: IFluidHandle = parseHandles({ type: "__fluid_handle__", url: unreferencedId }, serializer);
 
-            assert(response !== undefined, `Expecting a response!`);
-            assert(response.status === 404);
-            assert(response.value === `Datastore removed by gc: ${unreferencedId}`);
+            // Note: if a user makes a request that looks like this, we will also think that the request is via handle
+            await assert.rejects(
```

Comment:
What about recovery in this case? It could be our bug or app's but in either case, they should be able to recover right?

Reply:
Recovery is covered in another test (Can un-tombstone datastores by storing a handle)